BeardedManStudios/Source/BMSByte.cs

```diff
@@ -380,150 +360,24 @@ public void Clear(bool raw = false)
 			if (raw)
 				Append(new byte[] { 1 });
 		}
-
+		
 		/// <summary>
 		/// Get an array of a specific type from the internally tracked byte array
 		/// </summary>
 		/// <param name="type">The type of array to be pulled</param>
 		/// <param name="o"></param>
 		/// <returns>The updated array of data</returns>
-		private Array GetArray(Type type, object o)
+		private Array GetArray<T>(object o)
 		{
-			if (type == typeof(sbyte))
-			{
-				((sbyte[])allowedTypes[typeof(sbyte)])[0] = (sbyte)o;
-				return allowedTypes[typeof(sbyte)];
-			}
-			else if (type == typeof(byte))
-			{
-				((byte[])allowedTypes[typeof(byte)])[0] = (byte)o;
-				return allowedTypes[typeof(byte)];
-			}
-			else if (type == typeof(char))
-			{
-				((char[])allowedTypes[typeof(char)])[0] = (char)o;
-				return allowedTypes[typeof(char)];
-			}
-			else if (type == typeof(bool))
+			Type type = typeof(T);
+			if (type == typeof(sbyte) || type == typeof(byte) || type == typeof(char) || type == typeof(bool) || type == typeof(short) || type == typeof(ushort) || type == typeof(int) ||
+				type == typeof(uint) || type == typeof(long) || type == typeof(ulong) || type == typeof(float) || type == typeof(double))
```

Comment:
You are missing typeof(string). Also I think it would be better to do a check comparison with a 'Contains' from a List of types, or an array of types to compare against rather than just manually typing out a bunch of else comparisons. If you still want to pursue this route then I recommend making a function of this check rather than putting it all in this singular if check for better readability.

Reply:
string is never used, there is:
```
if (obj is string)
    Buffer.BlockCopy(Encoding.UTF8.GetBytes(((string)obj)), 0, byteArr, index, count);
else
    Buffer.BlockCopy(GetArray<T>(obj), 0, byteArr, index, count);
```